cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+            "network_used_cost": network_used_cost,
+            "disk_used_cost": disk_used_cost,
+            "storage_used_cost": storage_used_cost}
+
+
+@pytest.yield_fixture(scope="module")
+def chargeback_report_custom(vm_ownership, assign_custom_rate, interval):
+    """Create a Chargeback report based on a custom rate; Queue the report"""
+    owner = vm_ownership
+    data = {
+        'menu_name': interval,
+        'title': interval,
+        'base_report_on': 'Chargeback for Vms',
+        'report_fields': ['Memory Used', 'Memory Used Cost', 'Owner',
+        'CPU Used', 'CPU Used Cost',
+        'Disk I/O Used', 'Disk I/O Used Cost',
+        'Network I/O Used', 'Network I/O Used Cost',
+        'Storage Used', 'Storage Used Cost'],
+        'filter': {
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'
+        }
+    }
+    report = CustomReport(is_candu=True, **data)
+    report.create()
+
+    logger.info('Queuing chargeback report for {} rate'.format(interval))
+    report.queue(wait_for_finish=True)
+
+    if not list(report.get_saved_reports()[0].data.rows):
+        pytest.skip('Empty report')
+    else:
+        yield list(report.get_saved_reports()[0].data.rows)
+
+    if report.exists:
+        report.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def new_compute_rate(interval):
+    """Create a new Compute Chargeback rate"""
+    desc = 'custom_{}'.format(interval)
+    compute = rates.ComputeRate(description=desc,
+                fields={'Used CPU':
+                       {'per_time': interval, 'variable_rate': '720'},
+                       'Used Disk I/O':
+                       {'per_time': interval, 'variable_rate': '720'},
+                       'Used Network I/O':
+                       {'per_time': interval, 'variable_rate': '720'},
+                       'Used Memory':
+                       {'per_time': interval, 'variable_rate': '720'}})
+    compute.create()
+    storage = rates.StorageRate(description=desc,
+                fields={'Used Disk Storage':
+                        {'per_time': interval, 'variable_rate': '720'}})
+    storage.create()
+
+    yield desc
+
+    compute.delete()
+    storage.delete()
+
+
+def test_validate_cpu_usage_cost(chargeback_costs_custom, chargeback_report_custom,
+        interval, provider, soft_assert):
+    """Test to validate CPU usage cost reported in chargeback reports.
+    The cost reported in the Chargeback report should be approximately equal to the
+    cost estimated in the chargeback_costs_custom fixture.
+    """
+    for groups in chargeback_report_custom:
+        if not groups["CPU Used Cost"]:
+            pytest.skip('missing column in report')
+        else:
+            estimated_cpu_usage_cost = chargeback_costs_custom['cpu_used_cost']
+            cost_from_report = groups["CPU Used Cost"]
+            cost = re.sub(r'[$,]', r'', cost_from_report)
```

Comment:
REQUIRED: please use string's .replace instead

Reply:
Yes, with `.replace()` you don't need to import `re`.